### PR TITLE
[flutter_markdown] Use Text.rich to replace RichText in Flutter Markdown

### PR DESCRIPTION
**Pull request by leiatfly**
With this PR, a client can use SelectionArea to handle selected text in Markdown. 

When setting `selectable` to false, the Markdown widget renders Text, which can be wrapped in SelectionArea for client-defined behavior (i.e., the Markdown widget no longer consumes the selection event). 

When setting `selectable` to true, the Markdown widget handles the selection behavior, as before this change.

This fixes issue #107073 and is related to issue #113530.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

**Comment by leiatcornell** [comment]
Hi, I've now modified all the affected tests and have all checks passed. Could you please merge the PR?

**Comment by domesticmouse** [comment]
PTAL @stuartmorgan

File: packages/flutter_markdown/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 0.6.18+4
```
Comment (stuartmorgan) [inline packages/flutter_markdown/CHANGELOG.md:1]: Non-breaking behavior changes should be minor, not bugfix, version updates, as described in the semver spec, so this should be 0.6.19.

File: packages/flutter_markdown/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
...
+
+* Uses Text.rich to replace RichText so the widget can work with `SelectionArea` when `selectable` set to false.
```
Comment (stuartmorgan) [inline packages/flutter_markdown/CHANGELOG.md:3]: Please use backticks consistently for all code references.

File: packages/flutter_markdown/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
...
+* Updates affected tests.
```
Comment (stuartmorgan) [inline packages/flutter_markdown/CHANGELOG.md:4]: All changes update affected tests; that's not client-facing, so doesn't go in the CHANGELOG.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
> With this PR, a client can use SelectionArea to handle selected text in Markdown.
> 
> [...]
> 
> This is related to issue #107073 and issue #113530.

Does it fix 107173? If so, please update the description to say so explicitly.

> - [x] I added new tests to check the change I am making, or this PR is [test-exempt](https://github.com/flutter/flutter/wiki/Tree-hygiene#tests).

Where is the new test that validates that the change described in the PR is actually working (and does not regress in the future)? I just see boilerplate changes to existing tests.

File: packages/flutter_markdown/CHANGELOG.md (hunk 1 above)
Comment (leiatcornell) [reply packages/flutter_markdown/CHANGELOG.md:1]: Done.

File: packages/flutter_markdown/CHANGELOG.md (hunk 2 above)
Comment (leiatcornell) [reply packages/flutter_markdown/CHANGELOG.md:3]: Done.

File: packages/flutter_markdown/CHANGELOG.md (hunk 3 above)
Comment (leiatcornell) [reply packages/flutter_markdown/CHANGELOG.md:4]: Done.

**Comment by leiatcornell** [comment]
Thanks for the review, PTAL @stuartmorgan

File: packages/flutter_markdown/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+## 0.6.19
+
+* Uses `Text.rich` to replace `RichText` so the widget can work with `SelectionArea` when `selectable` set to false.
```
Comment (stuartmorgan) [inline packages/flutter_markdown/CHANGELOG.md:3]: Nit: is set to false

Comment (stuartmorgan) [inline packages/flutter_markdown/CHANGELOG.md:3]: Nit: Replaces `RichText` with `Text.rich`

**Review by stuartmorgan — APPROVED** [review]
Minor changelog nits, but otherwise looks good.

**Comment by stuartmorgan** [comment]
@peixinli In the future, please use the autosubmit label; manual merges bypass checks and load balancing, and are strongly discouraged in Flutter.

**Comment by peixinli** [comment]
> @peixinli In the future, please use the autosubmit label; manual merges bypass checks and load balancing, and are strongly discouraged in Flutter.

Sorry about that. Will do next time! Let me know if you think we need to revert.

**Comment by stuartmorgan** [comment]
No need for a revert, just something to minimize in the future.

**Pull request merged**